sdk/docs/iot/README.md

```diff
@@ -174,6 +174,126 @@ void my_telemetry_func()
 }
 ```
 
+### IoT Hub Pseudocode Implementation 
+
+Below is a pseudocode implementation for using the IoT Hub Client SDK. This is meant to guide users in incorporating their MQTT stack with the IoT Hub Client SDK. Note for simplicity reasons, this code will not compile. Ideally, guiding principles can be inferred from reading through this snippet to create an IoT solution.
+
+```C
+#include <az/core/az_result.h>
+#include <az/core/az_span.h>
+#include <az/iot/az_iot_hub_client.h>
+
+az_iot_hub_client my_client;
+static az_span my_iothub_hostname = AZ_SPAN_LITERAL_FROM_STR("<your hub fqdn here>");
+static az_span my_device_id = AZ_SPAN_LITERAL_FROM_STR("<your device id here>");
+
+//Make sure to size the buffer to fit the user name (100 is an example)
+static char my_mqtt_user_name[100];
+static size_t my_mqtt_user_name_length;
+
+//Make sure to size the buffer to fit the client id (16 is an example)
+static char my_mqtt_client_id[16];
+
+//This assumes an X509 Cert. SAS keys may also be used.
+static const char my_device_cert[]= "-----BEGIN CERTIFICATE-----abcdefg-----END CERTIFICATE-----";
+
+static char telemetry_topic[128];
+static char telemetry_payload[] = "Hello World";
+
+void handle_iot_message(mqtt_client_message* msg);
+
+int main()
+{
+  //Get the default IoT Hub options
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+
+  //Initialize the client with hostname, device id, and options
+  az_iot_hub_client_init(&my_client, my_iothub_hostname, my_device_id, &options);
+
+  //Get the MQTT user name to connect
+  az_iot_hub_client_get_user_name(&my_client, my_mqtt_user_name, 
+                sizeof(my_mqtt_user_name), NULL);
+
+  //Get the MQTT client id to connect
+  az_iot_hub_client_get_client_id(&my_client, my_mqtt_client_id, 
+                sizeof(my_mqtt_client_id), &my_mqtt_client_id_length);

```

Comment:
my_mqtt_client_id_length not defined..needed?

Reply:
changed to NULL